java/arcs/core/policy/PolicyTranslation.kt

```diff
@@ -0,0 +1,37 @@
+package arcs.core.policy
+
+import arcs.core.analysis.RecipeGraph
+
+/**
+ * Translates the given [Policy] into dataflow analysis checks and claims on the given
+ * [RecipeGraph], returning a modified copy of the graph.
+ */
+fun applyPolicy(policy: Policy, graph: RecipeGraph): RecipeGraph {
+    checkEgressParticles(policy, graph)
+    // TODO(b/157605232): Modify the graph and attach checks and claims.
+    return graph
+}
+
+private fun checkEgressParticles(policy: Policy, graph: RecipeGraph) {
+    val invalidEgressParticles = graph.particleNodes
+        .map { it.particle.spec }
+        .filter { !it.isolated && it.name != policy.egressParticleName }
```

Comment:
Any reason the following won't work?
```suggestion
        .filter { 
             val spec = it.particle.spec
             !spec.isolated && spec.name != policy.egressParticleName 
        }
```

Reply:
It would work, but I want the result to be a list of `ParticleSpec` not `Node.Particle` (for the next step).

Reply:
Oh, never mind then. You would have to do a map anyway then.